.github/autobuild/get_build_vars.py

```diff
@@ -60,8 +47,9 @@ def set_github_variable(varname, varval):
 
 
 jamulus_pro_version = get_version_from_jamulus_pro()
-write_changelog(jamulus_pro_version)
-build_version = get_build_version(jamulus_pro_version)
+set_github_variable("JAMULUS_PRO_VERSION", jamulus_pro_version)
+build_type, build_version = get_build_version(jamulus_pro_version)
+print(f'building a version of type "{build_type}": {build_version}')
 
 fullref = os.environ['GITHUB_REF']
 publish_to_release = bool(re.match(r'^refs/tags/r\d+_\d+_\d+\S*$', fullref))
```

Comment:
OK, can we _either_ do this or do what line 36 (`if "dev" in jamulus_pro_version:`) does, rather than have both ways in use.

I guess, ideally that means another function `get_release_type()` that just returns whether this is a `release` or not.  It can be boolean, to allow the caller to decide how to use the result - but if we're using the tag (as here) to drive what gets triggered in the automation, then the scripts (that get triggered) should also just use the tag.

So
- do we need to differentiate between `GITHUB_REF` version and `Jamulus.pro` version?
- if so, we should be consistent and use `GITHUB_REF` at least to drive whether a build is a release or not
- if not, we should be completely consistent and use one or other - everywhere

Reply:
> OK, can we _either_ do this or do what line 36 (`if "dev" in jamulus_pro_version:`) does, rather than have both ways in use.

I don't quite see yet how one could replace the other.

> I guess, ideally that means another function `get_release_type()` that just returns whether this is a `release` or not. It can be boolean, to allow the caller to decide how to use the result

That would be possible, for sure.

>  but if we're using the tag (as here) to drive what gets triggered in the automation, then the scripts (that get triggered) should also just use the tag.
> 
>  * do we need to differentiate between `GITHUB_REF` version and `Jamulus.pro` version?

With the current design, assumptions and constraints, we have to differentiate, I think. `GITHUB_REF`(= tag) might be different (e.g. for pre-releases) or not existing at all (branches/master).

>  * if so, we should be consistent and use `GITHUB_REF` at least to drive whether a build is a release or not

I think this should be the case already. We use the workflow variable `PUBLISH_TO_RELEASE` to drive the build. This variable is solely driven by `GITHUB_REF`.

>  * if not, we should be completely consistent and use one or other - everywhere

I think there's certainly room for improvement, but it needs thorough checking to avoid breaking existing use cases (such as local builds). In addition, I don't think this PR introduces any additional inconsistencies. It might make existing strangeness more obvious (which is a good thing, IMO). Therefore, I'm a bit hesitant to squeeze in additional larger changes here.

Reply:
My concern is `get_build_version()` uses a different value.  The tag, `Jamulus.pro` and `ChangeLog` could all have different versions.

Given we have three sources for the version, we really need automation to make sure they're aligned, at least: i.e. error out if things look wrong.  But I agree, that's not in scope here.  For that, it sounds like (and I don't fully understand how this works):
* If the tag is set, we should use that to determine build type and version and that source should overrule others.
* ChangeLog and Jamulus.pro "target" versions should match - and, if set, match the tag build type and version.
* If the tag wasn't set, build type and version can then come from ChangeLog and Jamulus.pro, as needed.

Reply:
Yep. I've split that out into #2662 to track that.